crates/nu-engine/src/eval.rs

```diff
@@ -875,6 +875,15 @@ pub fn eval_block(
     redirect_stdout: bool,
     redirect_stderr: bool,
 ) -> Result<PipelineData, ShellError> {
+    if let Some(recursion) = block.recursion {
+        if recursion {
+            if *stack.recursion_limit == 50 {
```

Comment:
Is 50 what rust defaults to? I thought you said it was a higher number?

Reply:
It's default depend on architecture, 
On Linux it's 128 .
I want to extend this section to working and throw error but 50 is minimal limit for all architecture

Reply:
ok, thanks for the info.

Reply:
> ok, thanks for the info.

If it has possibility to merge ,  would be great , some work on `stack` will conflict on this by working on `ls_color` on my next pull request which is uploading now ...
Thanks again @fdncred

Reply:
Can you please add a comment explaining why you picked 50?

I like to see comments for "magic numbers"; in the future somebody is going to wonder why this was set to 50. Even something like `// picked 50 arbitrarily, should work on all architectures` will help that person out.